components/ws-daemon/cmd/run.go

```diff
@@ -130,32 +74,25 @@ var runCmd = &cobra.Command{
 
 		go config.Watch(configFile, dmn.ReloadConfig)
 
-		// run until we're told to stop
-		sigChan := make(chan os.Signal, 1)
-		signal.Notify(sigChan, os.Interrupt, syscall.SIGTERM)
-		log.Info("🧫 ws-daemon is up and running. Stop with SIGINT or CTRL+C")
-		<-sigChan
-		server.Stop()
-		err = dmn.Stop()
+		err = srv.ListenAndServe()
 		if err != nil {
-			log.WithError(err).Error("cannot shut down gracefully")
+			log.Fatal(err)
 		}
-		log.Info("Received SIGINT - shutting down")
 	},
 }
 
 func init() {
 	rootCmd.AddCommand(runCmd)
 }
 
-func grpcProbe(tlsConfig config.AddrTLS) func() error {
+func grpcProbe(cfg baseserver.ServerConfiguration) func() error {
 	return func() error {
 		secopt := grpc.WithInsecure()
-		if tlsConfig.TLS != nil && tlsConfig.TLS.Certificate != "" {
+		if cfg.TLS != nil && cfg.TLS.CertPath != "" {
 			tlsConfig, err := common_grpc.ClientAuthTLSConfig(
-				tlsConfig.TLS.Authority, tlsConfig.TLS.Certificate, tlsConfig.TLS.PrivateKey,
+				cfg.TLS.CAPath, cfg.TLS.CertPath, cfg.TLS.KeyPath,
 				common_grpc.WithSetRootCAs(true),
-				common_grpc.WithServerName("wsdaemon"),
+				common_grpc.WithServerName(grpcServerName),
 			)
 			if err != nil {
 				return xerrors.Errorf("cannot load ws-daemon certificate: %w", err)
```

Comment:
```suggestion
				return xerrors.Errorf("cannot load %s certificate: %w", grpcServerName, err)
```

Reply:
```suggestion
				return fmt.Errorf("cannot load ws-daemon certificate: %w", err)
```
Actually this, because `xerrors.Errorf` is deprecated.